internal/network/data_sender.go

```diff
@@ -231,28 +230,27 @@ func (ds *dataSender) SendDeviceNetworkData(data models.DeviceNetworkData) error
 	return nil
 }
 
-func (ds *dataSender) SendCanDumpData(data models.CanDumpData) error {
-	if data.Timestamp == 0 {
-		data.Timestamp = time.Now().UTC().UnixMilli()
-	}
-	ce := shared.CloudEvent[models.CanDumpData]{
-		ID:          ksuid.New().String(),
-		Source:      "aftermarket/device/canbus/dump",
-		SpecVersion: "1.0",
-		Subject:     ds.ethAddr.Hex(),
-		Time:        time.Now().UTC(),
-		Type:        "zone.dimo.aftermarket.canbus.dump",
-		DataSchema:  "dimo.zone.status/v2.0",
-		Data:        data,
-	}
-	println("Sending can dump data: (payload)")
+// SendCanDumpData sends a byte array, compressed, to mqtt candump topic
+func (ds *dataSender) SendCanDumpData(data json.RawMessage) error {
+	ce := shared.CloudEvent[json.RawMessage]{
+		ID:             ksuid.New().String(),
+		Source:         "aftermarket/device/canbus/dump",
+		SpecVersion:    "1.0",
+		Subject:        ds.ethAddr.Hex(),
+		Time:           time.Now().UTC(),
+		Type:           "zone.dimo.aftermarket.canbus.dump",
+		DataSchema:     "dimo.zone.status/v2.0",
+		Data:           data,
+		VehicleTokenID: uint32(ds.vehicleInfo.TokenID),
+	}
+	ds.logger.Info().Msgf("Sending can dump data: %+v", ce)
 	payload, err := json.Marshal(ce)
-	println(payload)
 	if err != nil {
 		return errors.Wrap(err, "failed to marshall cloudevent")
 	}
 
-	err = ds.sendPayload(canDumpTopic, payload, false)
+	candump := fmt.Sprintf(ds.mqtt.Topics.Candump, ce.Subject)
```

Comment:
are we changing topic name from protocol/canbus/dump to devices/0x/protocol/canbus/dump ?

Reply:
same name, thing is that before it used the built in mosquitto which prepended the devices/
i checked EMQX and topic i think should match, maybe double check

Reply:
oh, you are right , I forgot about it